chains/manager.go

```diff
@@ -531,15 +531,10 @@ func (m *manager) buildChain(chainParams ChainParameters, sb subnets.Subnet) (*c
 		}
 	}
 
-	var vdrs validators.Set // Validators validating this blockchain
-	var hasValidators bool
-	if m.SybilProtectionEnabled {
-		vdrs, hasValidators = m.Validators.Get(chainParams.SubnetID)
-	} else { // Sybil protection is disabled. Every peer validates every subnet.
-		vdrs, hasValidators = m.Validators.Get(constants.PrimaryNetworkID)
-	}
-	if !hasValidators {
-		return nil, fmt.Errorf("couldn't get validator set of subnet with ID %s. The subnet may not exist", chainParams.SubnetID)
+	vdrs := m.Validators
+	if !m.SybilProtectionEnabled {
```

Comment:
Can we completely get rid of this if-check? I think we can get rid of this + the one in the benchlist if we directly set `OverriddenManager` in `Node` [here](https://github.com/ava-labs/avalanchego/blob/8d2c4f211e9f408fe382e7c5a52cac9435d6f6b3/node/node.go#L1273).

Reply:
hmm that's a very good idea/cleanup. the only place makes me worry is `NetworkConfig`: https://github.com/ava-labs/avalanchego/blob/dev/node/node.go#L367
I think it should still be OK since every call for validators will be made for primary network in networking(which fits well with Sybil protection disabled) but better to ask @StephenButtolph.

Reply:
Created https://github.com/ava-labs/avalanchego/pull/2199

Reply:
I think this change is safe to make as-is but I'm down to this as a follow-up since these if-checks exist in the code as-is.